### PR TITLE
feat: scaletest: scale down nodegroups by default

### PR DESCRIPTION
**Pull request by johnstcn**
This PR modifies the scaletest terraform to allow scaling down the cluster by setting `-var state=stopped`. This should hopefully save some time performing scaletests as we no longer have to wait for the cloudsql database etc. to be created.

By default, `scaletest.sh` will scale down the nodepools unless the `--destroy` argument is passed.

### CODE REVIEW

**Review by mtojek — APPROVED** [review]
Why do I have a feeling that it performs more than just starting with `nodecount = 0` :)

**Comment by johnstcn** [comment]
> Why do I have a feeling that it performs more than just starting with `nodecount = 0` :)

There were some... interesting gotcha's. The main thing I found is that when you try to delete a namespace in GKE with no active nodepools is that the namespace will hang in the `Deleting` state, probably due to some finalizers that haven't run. 

I spent a while chasing the 'right' way to ignore the resource deletion in Terraform before deciding to just use a null_resource for creating the namespaces and moving on with my life.

File: scaletest/scaletest.sh [hunk 1]
```diff
@@ -151,7 +156,21 @@ else
...
+		break
+	fi
+	if attempt -eq $max_attempts; then
+		echo
+		echo "Coder deployment failed to become ready in time!"
+		exit 1
+	fi
+	echo "Coder deployment not ready yet (${attempt}/${max_attempts}), sleeping 3 seconds"
+	maybedryrun "$DRY_RUN" sleep 3
+done
```
Comment (johnstcn) [inline scaletest/scaletest.sh:173]: review: there is a race condition between the `rollout status` returning true and the service actually becoming ready; so I'm just going back to curl :-)

File: scaletest/terraform/gcp_cluster.tf [hunk 2]
```diff
@@ -124,3 +128,25 @@ resource "google_container_node_pool" "misc" {
...
+    EOF
+  }
+
+  provisioner "local-exec" {
+    when    = destroy
+    command = <<EOF
+      rm -f ${self.triggers.path}
+    EOF
+  }
+}
```
Comment (johnstcn) [inline scaletest/terraform/gcp_cluster.tf:152]: review: this was previously being created when applying prom monitoring manifests; moved it to its own resource declaration here.

**Pull request merged**